src/zh/cartoonmad/src/eu/kanade/tachiyomi/extension/zh/cartoonmad/CartoonMad.kt

```diff
@@ -0,0 +1,151 @@
+package eu.kanade.tachiyomi.extension.zh.cartoonmad
+
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.network.POST
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.ParsedHttpSource
+import eu.kanade.tachiyomi.util.asJsoup
+import okhttp3.FormBody
+import okhttp3.Interceptor
+import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.Response
+import okhttp3.ResponseBody.Companion.asResponseBody
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+
+class CartoonMad : ParsedHttpSource() {
+    override val baseUrl = "https://www.cartoonmad.com"
+    override val lang: String
+        get() = "zh"
+    override val name: String
+        get() = "動漫狂"
+    override val supportsLatest: Boolean
+        get() = true
+
+    override val client: OkHttpClient = network.cloudflareClient.newBuilder()
+        .addInterceptor(::handleCharsetInterceptor)
+        .build()
+
+    fun handleCharsetInterceptor(chain: Interceptor.Chain): Response {
+        val request = chain.request()
+        val response = chain.proceed(request)
+        if (request.url.encodedPath.contains("/comic/")) {
```

Comment:
Check the specific path segment instead.
```suggestion
        if (request.url.pathSegments.getOrNull(0) == "comic") {
```

Reply:
In the mobile , the path is `/m/comic/`, while in the PC version, the path is `/comic/`. It’s more convenient to use String#contains for this.